_delphi_utils_python/tests/validator/test_dynamic.py

```diff
@@ -465,3 +465,38 @@ def test_source_api_overlap(self):
 
         assert len(report.raised_warnings) == 2
         assert report.raised_warnings[0].check_name == "check_positive_negative_spikes"
+
+class TestDateComparison:
+    params = {
+        "common": {
+            "data_source": "",
+            "span_length": 1,
+            "end_date": "2020-09-02"
+        }
+    }
+
+    def test_date_comparison_by_type(self):
+        validator = DynamicValidator(self.params)
+        report = ValidationReport([])
+
+        ref_val = [30, 30.28571429, 30.57142857, 30.85714286, 31.14285714,
+                   31.42857143, 31.71428571, 32, 32, 32.14285714,
+                   32.28571429, 32.42857143, 32.57142857, 32.71428571,
+                   32.85714286, 33, 33, 33, 33, 33, 33, 33, 33,
+                   33, 33, 33, 33.28571429, 33.57142857, 33.85714286, 34.14285714]
```

Comment:
suggestion: remove unnecessary complexity

Do the exact values & length matter for this, or can we simplify for easier reading?

```suggestion
        ref_val = [30, 30, 30]
```